pkg/spanconfig/systemtarget.go

```diff
@@ -43,15 +43,36 @@ type SystemTarget struct {
 
 // MakeSystemTarget constructs, validates, and returns a new SystemTarget.
 func MakeSystemTarget(
-	sourceTenantID roachpb.TenantID, targetTenantID *roachpb.TenantID,
+	sourceTenantID roachpb.TenantID, targetTenantID roachpb.TenantID,
```

Comment:
For future PRs: worth reconsidering IMO these constructors, in addition to MakeClusterTarget. My preference:
```
MakeClusterTarget(roachpb.TenantID) // takes in what you call today "source
MakeTenantOverrideTarget(roachpb.TenantID) // doesn't take in the source, has to be system tenant, but takes in dest
```

Pick whatever names you want for these.

Reply:
What we ended up with for the time being is:

```
MakeClusterTarget() SystemTarget // returns a system target with only the sourceTarget set to the system tenant ID 
MakeTenantTarget(source, target roachpb.TenantID)
```

Then the SQLTranslator will:
```
	// Aggregate cluster target protections for the tenant.
	clusterProtections := ptsStateReader.getProtectionPoliciesForCluster()
	if len(clusterProtections) != 0 {
		var systemTarget spanconfig.SystemTarget
		var err error
		if sourceTenantID == roachpb.SystemTenantID {
			systemTarget = spanconfig.MakeClusterTarget()
		} else {
			systemTarget, err = spanconfig.MakeTargetTenant(sourceTenantID, sourceTenantID)
			if err != nil {
				return nil, err
			}
		}
		clusterSystemRecord := spanconfig.Record{
			Target: spanconfig.MakeTargetFromSystemTarget(systemTarget),
			Config: roachpb.SpanConfig{GCPolicy: roachpb.GCPolicy{ProtectionPolicies: clusterProtections}}}
		records = append(records, clusterSystemRecord)
	}

	// Aggregate tenant target protections.
	tenantProtections := ptsStateReader.getProtectionPoliciesForTenants()
	for _, protection := range tenantProtections {
		tenantProtection := protection
		tenantSystemRecord := spanconfig.Record{
			Target: spanconfig.MakeTargetFromSystemTarget(spanconfig.SystemTarget{
				SourceTenantID: sourceTenantID,
				TargetTenantID: &tenantProtection.tenantID,
			}),
			Config: roachpb.SpanConfig{GCPolicy: roachpb.GCPolicy{
				ProtectionPolicies: tenantProtection.protections}},
		}
		records = append(records, tenantSystemRecord)
	}
```

Reply:
But we could make it simpler without complicating the client side:

```go
	// Aggregate cluster target protections for the tenant.
	clusterProtections := ptsStateReader.getProtectionPoliciesForCluster()
	if len(clusterProtections) != 0 {
		clusterSystemRecord := spanconfig.Record{
			Target: spanconfig.MakeTargetFromSystemTarget(spanconfig.MakeClusterTarget(sourceTenantID)),
			Config: roachpb.SpanConfig{GCPolicy: roachpb.GCPolicy{ProtectionPolicies: clusterProtections}}}
		records = append(records, clusterSystemRecord)
	}

	// Aggregate tenant target protections.
	tenantProtections := ptsStateReader.getProtectionPoliciesForTenants()
	for _, protection := range tenantProtections {
		tenantProtection := protection
		tenantSystemRecord := spanconfig.Record{
			Target: spanconfig.MakeTargetFromSystemTarget(spanconfig.MakeTenantTarget(tenantProtection.tenantID)),
			Config: roachpb.SpanConfig{GCPolicy: roachpb.GCPolicy{
				ProtectionPolicies: tenantProtection.protections}},
		}
		records = append(records, tenantSystemRecord)
	}
```

Reply:
If we want to simplify the client side logic, it might be worth considering changing the `ptsStateReader` to return `ptpb.Targets` instead of protection policies like it does today. Then, we could write conversions to/from `ptpb.Targets` to `spanconfig.Targets`.

Reply:
It's not about simplifying client code, it's about concepts. We have a cluster concept in backup/pts, and we have another cluster concept here that means a wholly different thing. They both have the same MakeClusterTarget signature too. It's a bit difficult to see why.